### PR TITLE
Validate application_id and application_version

### LINKED ISSUE #157: Validate application_id and application_version
Senza's TaupageAutoScalingGroup should validate application_id and application_version as we do in Kio (https://github.com/zalando-stups/kio/blob/master/resources/api/kio-api.yaml#L482).

* application_id must match pattern: "^[a-z][a-z0-9-]*[a-z0-9]$"
* application_version must match pattern: "^[A-Za-z0-9](?:[A-Za-z0-9._-]*[A-Za-z0-9])?$"

Background: sometimes users accidentally put the full Docker image (pierone.example.org/../..:123) into application_version

### PR DESCRIPTION
**Pull request by jmcs**
Fixes #157

### CODE REVIEW

**Comment by prayerslayer** [comment]
looks good

File: senza/components/taupage_auto_scaling_group.py [hunk 1]
```diff
@@ -14,6 +14,24 @@
 
 _AWS_FN_RE = re.compile(r"('[{]{2} (.*?) [}]{2}')", re.DOTALL)
 
+# from kio OpenAPI yaml
+APPLICATION_ID_RE = re.compile(r"^[a-z][a-z0-9-]*[a-z0-9]$")
+APPLICATION_VERSION_RE = re.compile(r"^[A-Za-z0-9](?:[A-Za-z0-9._-]*[A-Za-z0-9])?$")
```
Comment (rafaelcaricio) [inline senza/components/taupage_auto_scaling_group.py]: This RE does have unnecessary stuff also it matches something like ``1..3.....4`` as a valid version.
 I would recommend using this simplified version ``^[A-Za-z0-9]+([._-]?[A-Za-z0-9]+)*$``

Comment (jmcs) [inline senza/components/taupage_auto_scaling_group.py]: I'm using the same on as kio.

Comment (rafaelcaricio) [inline senza/components/taupage_auto_scaling_group.py]: Should we propagate the mistake or fix it? :poop:
I will send a PR to kio.

Comment (hjacobs) [inline senza/components/taupage_auto_scaling_group.py]: Why is the "mistake" a problem? I personally don't care, it's only about valid character ranges and that it does not start/end with a symbol, having ten dots after each other is stupid, but not illegal.

**Comment by hjacobs** [comment]
:+1:

File: senza/components/taupage_auto_scaling_group.py [hunk 2]
```diff
@@ -14,6 +14,24 @@
...
+
+
+def check_application_id(app_id: str) -> bool:
+    if not APPLICATION_ID_RE.match(app_id):
+        raise click.UsageError('Application id must satisfy regular '
+                               'expression pattern "{}"'.format(APPLICATION_ID_RE.pattern))
+    return True
```
Comment (hjacobs) [inline senza/components/taupage_auto_scaling_group.py]: do we need the return? we could get rid of the two lines :smirk:

File: senza/components/taupage_auto_scaling_group.py (hunk 1 above)
Comment (rafaelcaricio) [inline senza/components/taupage_auto_scaling_group.py]: If everyone is fine with the "different" version pattern. OK! 👍

Just wanted to express my opinion -- If it is to validate versions, lets validate it properly.

**Comment by rafaelcaricio** [comment]
👍

File: senza/components/taupage_auto_scaling_group.py (hunk 1 above)
Comment (rafaelcaricio) [inline senza/components/taupage_auto_scaling_group.py]: https://github.com/zalando-stups/kio/pull/66

**Pull request merged**